libslide/src/grammar.rs

```diff
@@ -133,14 +136,32 @@ impl fmt::Display for Expr {
     }
 }
 
-#[derive(PartialEq, Clone, Copy, Hash)]
+#[derive(PartialEq, Eq, Clone, Copy, Hash, Debug)]
 pub enum BinaryOperator {
-    Plus,
-    Minus,
-    Mult,
-    Div,
-    Mod,
-    Exp,
+    Plus = 1,
+    Minus = 2,
+    Mult = 10,
+    Div = 11,
+    Mod = 12,
+    Exp = 20,
+}
+
```

Comment:
why did we set defaults?

Reply:
So that we can do partial comparison simply. Take a look at the `PartialOrd` impl

Reply:
Ahh i see. I think we should add comments here for that.